apps/meteor/ee/app/livechat-enterprise/server/api/lib/priorities.ts

```diff
@@ -53,13 +53,29 @@ export async function updatePriority(
 		}),
 	};
 
-	const created = await LivechatPriority.findOneAndUpdate(query, update, {
-		returnDocument: 'after',
-	});
+	try {
+		if (data.name) {
+			const priority = await LivechatPriority.findOne({ name: new RegExp(`^${escapeRegExp(data.name)}$`, 'i') });
+			if (priority && priority._id !== _id) {
+				throw new Error('error-duplicate-priority-name');
+			}
+		}
 
-	if (!created.ok || !created.value) {
-		throw Error('Error updating priority');
-	}
+		const created = await LivechatPriority.findOneAndUpdate(query, update, {
+			returnDocument: 'after',
+		});
+
+		if (!created.ok || !created.value) {
+			throw Error('Error updating priority');
+		}
 
-	return created.value;
+		return created.value;
+	} catch (error: any) {
+		// check if its a duplicate key error
+		if (error?.code === 11000) {
+			throw new Error('error-duplicate-priority-name');
+		}
```

Comment:
On L 59 You're checking there's no priority with same name, we can remove this check and:
1. Log the error
2. Throw a generic error of "priority cannot be stored" or smth